File: x2023/python/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    text = get()
    l = countletters(text)
    w = countwords(text)
    se = countsentences(text)
    L = (l / w) * 100
    S = (se / w) * 100
    index = 0.0588 * L - 0.296 * S - 15.8  # calculating index by the given formula
    if round(index) >= 1 and round(index) < 16:
        print(f"Grade {round(index)}")
    elif round(index) >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


# Function to take input from user
def get():
    text = get_string("Text: ")
    while len(text) < 2:
        text = get_string("Text: ")
    return text


# FUnction to count letters
def countletters(text):
    c = 0
    for i in range(len(text)):
        if text[i].isalpha() == True:
            c += 1
    return c


# Function to count words
def countwords(text):
    c = 1  # taking c is equal to 1 because no of words is one more than no. of spaces
    for i in range(len(text)):
        if text[i] == " ":
            c += 1
    return c


# Function to count sentences
def countsentences(text):
    c = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            c += 1
    return c


main()

File: x2023/c/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>
int countletters(string text);
int countwords(string text);
int countsentences(string text);
int main(void)
{
   string text=get_string("Text: ");
   double l=countletters(text);
   double w=countwords(text);
   double se=countsentences(text);
   double L=l/w*100; // average letters per words * 100
   double S=se/w*100; // average sentences per words *100
   double index = 0.0588 * L - 0.296 * S - 15.8;
   if((int) round(index)>=1 && (int) round(index)<16) //condition for printing grade by rounding index if it is between 1 to 16
   {
   printf("Grade %d\n",(int) round(index));
   }
   else if((int) round(index)>=16) //condition for printing grade 16+ when grade is more than or equal to 16
   {
      printf("Grade 16+\n");
   }
   else // else for printing "Before Grade 1" if grade is less than 1
   {
      printf("Before Grade 1\n");
   }
}
int countletters(string text) //function to count letters
{
   int s=strlen(text);
   int l=0;
   for(int i=0;i<s;i++)
   {
      if(isalpha(text[i]))
      {
      l++;
      }
   }
   return l;
}
int countwords(string text) //function to count words
{
  int s=strlen(text);
  int w=1;
  for(int i=0;i<s;i++)
  {
   if(isspace(text[i]))
   {
      w++;
   }
  }
  return w;
}
int countsentences(string text) //function to count no. of sentences
{
   int s=strlen(text);
   int se=0;
   for(int i=0;i<s;i++)
   {
     if(text[i]=='.' || text[i]=='!' || text[i]=='?')
     {
      se++;
     }
   }
   return se;
}